numpy_financial/_financial.py

```diff
@@ -391,13 +399,22 @@ def ipmt(rate, per, nper, pv, fv=0, when='end'):
     when = _convert_when(when)
     rate, per, nper, pv, fv, when = np.broadcast_arrays(rate, per, nper,
                                                         pv, fv, when)
+
     total_pmt = pmt(rate, nper, pv, fv, when)
-    ipmt = _rbl(rate, per, total_pmt, pv, when)*rate
-    try:
-        ipmt = np.where(when == 1, ipmt/(1 + rate), ipmt)
-        ipmt = np.where(np.logical_and(when == 1, per == 1), 0, ipmt)
-    except IndexError:
-        pass
+    ipmt = np.array(_rbl(rate, per, total_pmt, pv, when) * rate)
```

Comment:
Here and below the logic looks good to me. Having the extra comments is also appreciated. :+1: